### PR TITLE
Fix FxA GA bug for Sync device counts (Fixes #9929)

### LINKED ISSUE #9929: GA not reporting FxA Sync devices correctly
## Description

I have multiple active devices using Firefox Sync, however I don't see this being represented correctly in the data bedrock sends to GA. I think there's a bug in how we're reporting this data.

### Steps to reproduce

There's something not right in this piece of code: https://github.com/mozilla/bedrock/blob/master/media/js/base/core-datalayer-init.js#L36-L40

When I query `Mozilla.Client.getFxaDetails()` I get the following which is correct:

```javascript
{
  "firefox": true,
  "legacy": false,
  "mobile": false,
  "setup": true,
  "browserServices": {
    "sync": {
      "setup": true,
      "desktopDevices": 1,
      "mobileDevices": 1,
      "totalDevices": 0
    }
  }
}
```

However when I query `Mozilla.Analytics.formatFxaDetails(details)` (passing in the data above) I get a different result.

### Expected result

```javascript
{ 
  FxALogin: true, 
  FxASegment: "Desktop and Mobile Sync", 
  FxAMobileSync: true, 
  FxAMultiDesktopSync: false 
}
```

### Actual result

```javascript
{ 
  FxALogin: true, 
  FxASegment: "Logged in", 
  FxAMobileSync: "unknown", 
  FxAMultiDesktopSync: "unknown" 
}
```

### Environment

Firefox Nightly

### PR DESCRIPTION
**Pull request by alexgibson**
## Description
Make sure we're conforming to new standards defined in `Mozilla.Client.getFxaDetails()`.

## Issue / Bugzilla link
#9929

## Testing
- [x] Do my assertions make sense?
- [x] Are there any other edge cases I should consider?

### CODE REVIEW

File: tests/unit/spec/base/core-datalayer.js [hunk 1]
```diff
@@ -130,8 +130,14 @@ describe('core-datalayer.js', function() {
...
                 'mobile': false,
                 'setup': false,
-                'desktopDevices': false,
-                'mobileDevices': false
+                'browserServices': {
+                    'sync': {
+                        'setup': false,
+                        'desktopDevices': 'unknown',
+                        'mobileDevices': 'unknown',
+                        'totelDevices': 'unknown'
```
Comment (stephaniehobson) [inline tests/unit/spec/base/core-datalayer.js:138]: totel?

**Comment by alexgibson** [comment]
@stephaniehobson thanks updated

**Review by stephaniehobson — APPROVED** [review]
R+ 🛁
